tests/modes/apps/extensions/test_extension_remote.py

```diff
@@ -17,172 +16,84 @@ class TestExtensionRemote:
-            use_git=False, url="https://gitlab.com/user/repo", path="user/repo", protocol="https", host="gitlab.com"
-        )
+    def test_http_url_converts_http_to_https(self) -> None:
+        assert parse_extension_url("http://example.com/user/repo").remote_url.startswith("https://")
 
-    @patch("ulauncher.modes.extensions.extension_remote.which")
-    def test_codeberg_url_sanitization(self, mock_which: MagicMock) -> None:
-        mock_which.return_value = "/usr/bin/git"
-        result = parse_extension_url("https://codeberg.org/user/repo/wiki")
-
-        assert result == UrlParseResult(
-            use_git=False,
-            url="https://codeberg.org/user/repo",
-            path="user/repo",
-            protocol="https",
-            host="codeberg.org",
+    def test_url_sanitization(self) -> None:
+        assert parse_extension_url("  https://example.com/path  ").remote_url == "https://example.com/path"
+        assert parse_extension_url("https://github.com/user/repo").remote_url == "https://github.com/user/repo.git"
+        assert parse_extension_url("git@gitlab.com:user/repo.git").remote_url == "https://gitlab.com/user/repo.git"
+        assert (
+            parse_extension_url("https://github.com/user/repo/blob/master").remote_url
+            == "https://github.com/user/repo.git"
         )
+        assert parse_extension_url("https://gitlab.com/u/repo/issues").remote_url == "https://gitlab.com/u/repo.git"
+        assert parse_extension_url("https://codeberg.org/u/repo/wiki").remote_url == "https://codeberg.org/u/repo.git"
 
-    @patch("ulauncher.modes.extensions.extension_remote.which")
-    def test_git_extension_removal(self, mock_which: MagicMock) -> None:
-        mock_which.return_value = "/usr/bin/git"
-        result = parse_extension_url("https://github.com/user/repo.git")
-
-        assert result == UrlParseResult(
-            use_git=False, url="https://github.com/user/repo", path="user/repo", protocol="https", host="github.com"
-        )
+    def test_browser_url(self) -> None:
+        assert parse_extension_url("git@gitlab.com:user/repo.git").browser_url == "https://gitlab.com/user/repo"
 
-    @patch("ulauncher.modes.extensions.extension_remote.which")
-    def test_no_git_available(self, mock_which: MagicMock) -> None:
-        mock_which.return_value = None
-        result = parse_extension_url("https://example.com/user/repo")
+    def test_ext_id(self) -> None:
+        assert parse_extension_url("https://github.com/user/repo").ext_id == "com.github.user.repo"
+        assert parse_extension_url("https://example.co.uk/user/repo").ext_id == "uk.co.example.user.repo"
+        assert parse_extension_url("https://gitlab.com/user/repo/issues").ext_id == "com.gitlab.user.repo"
+        assert parse_extension_url("https://local/path/to/extension").ext_id == "local.path.to.extension"
+        assert parse_extension_url("https://localhost/extension").ext_id == "localhost.extension"
```

Comment:
New tests (copied from https://github.com/Ulauncher/Ulauncher/pull/1488/files#diff-7448673cfee888a244df38ce89fe74837aa4e231457092bd0ecdc4ace3b9a938R196-R205 excdept I changed the GitLab one).